orca-core/src/main/groovy/com/netflix/spinnaker/orca/pipeline/util/PackageInfo.groovy

```diff
@@ -139,10 +137,10 @@ class PackageInfo {
       }
 
       if (packageName) {
-        // At least one matched package must be present it the context.
-        if (!matched) {
-          notMachedPrefix = prefix
-          notMatchedFileExtension = fileExtension
+
+        // If it hasn't been possible to match a package and allowMissingPackageInstallation is false raise an exception.
+        if (!matched && !request.get("allowMissingPackageInstallation")) {
```

Comment:
If you decide to stick with throwing on the first miss, I think we can probably still drop `matched` entirely and move the `if (!request.get("allowMissingPackageInstallation"))` test within the `if (!triggerArtifact && !buildArtifact)` block.